### DISCUSSION TITLE
Call existing global functions

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [mdgrs-mei/PowerShellRun#61] [Ideas]
I am assuming that any function defined between the PSFunctionRegistration commands is categorized as a `Function`.  Is there a way to add to define a function using an existing global function? I know I can use `Add-PSRunScriptBlock` but there's no way to categorize it as a function.

**Comment by mdgrs-mei** [comment]
I assume you are using `Function` entry group and that's why you would like to add a script block or function to the `Function` category. What we can do for now is to get the `Function` entry group and add a script block to it:

```powershell
$functionEntryGroup = Add-PSRunEntryGroup -Name Functions -Category Function -PassThru

Add-PSRunScriptBlock -Name 'Test ScriptBlock' -EntryGroup $functionEntryGroup -ScriptBlock {
    'call your function here'
}
```

Implementing `Add-PSRunFunction` might be a natural addition in the future though.

**Reply by jdhitsolutions** [reply]
That's a reasonable workaround. I agree that adding `Add-PSRunFunction` makes the most sense conceptually in the long run.
